s3transfer/crt.py

```diff
@@ -556,22 +591,57 @@ def get_make_request_args(
         self, request_type, call_args, coordinator, future, on_done_after_calls
     ):
         recv_filepath = None
+        on_body = None
         send_filepath = None
         s3_meta_request_type = getattr(
             S3RequestType, request_type.upper(), S3RequestType.DEFAULT
         )
         on_done_before_calls = []
+        checksum_config = S3ChecksumConfig()
+
         if s3_meta_request_type == S3RequestType.GET_OBJECT:
-            final_filepath = call_args.fileobj
-            recv_filepath = self._os_utils.get_temp_filename(final_filepath)
-            file_ondone_call = RenameTempFileHandler(
-                coordinator, final_filepath, recv_filepath, self._os_utils
-            )
-            on_done_before_calls.append(file_ondone_call)
+            if isinstance(call_args.fileobj, str):
+                # fileobj is a filepath
+                final_filepath = call_args.fileobj
+                recv_filepath = self._os_utils.get_temp_filename(final_filepath)
+                file_ondone_call = RenameTempFileHandler(
+                    coordinator, final_filepath, recv_filepath, self._os_utils
+                )
+                on_done_before_calls.append(file_ondone_call)
+
+            elif call_args.fileobj is not None:
+                # fileobj is a file-like object
+                response_handler = _FileobjResponseHandler(call_args.fileobj)
+                on_body = response_handler.on_body
+
+            # Only validate response checksums when downloading.
+            # (upload responses also have checksum headers, but they're just an
+            # echo of what was in the request, an upload response's body is empty)
+            checksum_config.validate_response = True
+
         elif s3_meta_request_type == S3RequestType.PUT_OBJECT:
-            send_filepath = call_args.fileobj
-            data_len = self._os_utils.get_file_size(send_filepath)
-            call_args.extra_args["ContentLength"] = data_len
+            if isinstance(call_args.fileobj, str):
+                # fileobj is a filepath
+                send_filepath = call_args.fileobj
+                data_len = self._os_utils.get_file_size(send_filepath)
+                call_args.extra_args["ContentLength"] = data_len
+
+            elif call_args.fileobj is not None:
+                # fileobj is a file-like object
+                call_args.extra_args["Body"] = call_args.fileobj
+
+            # We want the CRT S3Client to calculate checksums, not botocore.
+            # Default to CRC32.
```

Comment:
This is different than the default TransferManager. With this change CRT is defaulting to CRC32 instead of Content-MD5. The CRT team got grumpy about how Content-MD5 currently works in our code and pushed me to do it this way, but we can still make Content-MD5 work if we need to.